cmd/geth/chaincmd.go

```diff
@@ -386,7 +389,13 @@ func dump(ctx *cli.Context) error {
 			if err != nil {
 				utils.Fatalf("could not create new state: %v", err)
 			}
-			fmt.Printf("%s\n", state.Dump())
+			excludeCode := ctx.GlobalIsSet(utils.ExcludeCodeFlag.Name)
+			excludeStorage := ctx.GlobalIsSet(utils.ExcludeStorageFlag.Name)
+			if ctx.GlobalIsSet(utils.IterativeOutputFlag.Name) {
```

Comment:
`GlobalIsSet` only returns whether the flag is set, not whether the boolean value is `true`. If I set `--iterative=false`, then `GlobalIsSet` will return true. You're looking for `GlobalBool`.